cpp/include/cudf/lists/lists_column_view.hpp

```diff
@@ -56,6 +56,8 @@ class lists_column_view : private column_view {
   using column_view::null_mask;
   using column_view::offset;
   using column_view::size;
+  using offset_type     = int32_t;
```

Comment:
This way we'll have a canary if either thing ever changes. 
```suggestion
  using offset_type     = int32_t;
  static_assert(std::is_same<offset_type, size_type>::value, "offset_type is expected to be the same as size_type.");
```